jetbrains-core/src/software/aws/toolkits/jetbrains/services/codemodernizer/model/CodeModernizerSessionContext.kt

```diff
@@ -162,135 +170,249 @@ data class CodeModernizerSessionContext(
+            buildlogBuilder.appendLine("$copyOutput")
+            if (copyOutput.exitCode == 0) {
+                LOG.info { "$mavenCommand copy-dependencies succeeded" }
+                buildlogBuilder.appendLine("$mavenCommand copy-dependencies succeeded")
+            } else {
+                LOG.error { "$mavenCommand copy-dependencies failed" }
+                buildlogBuilder.appendLine("$mavenCommand copy-dependencies failed")
+            }
+            return copyOutput
         }
 
         // 1. Try to execute Maven Wrapper Command
-        LOG.warn { "Executing ./mvnw" }
         var shouldTryMvnCommand = true
+        val mvnw = if (SystemInfo.isWindows) {
+            "./mvnw.cmd"
+        } else {
+            "./mvnw"
+        }
         try {
-            val mvnw = if (SystemInfo.isWindows) {
-                "./mvnw.cmd"
-            } else {
-                "./mvnw"
+            LOG.info { "Executing $mvnw" }
+            val installOutput = runInstallCommand(mvnw)
+            if (installOutput.exitCode != 0) {
+                LOG.error { "$mvnw install output: $installOutput" }
+                val error = "Maven Install: The exitCode should be 0 while it was ${installOutput.exitCode}"
+                CodetransformTelemetry.mvnBuildFailed(
+                    codeTransformSessionId = CodeTransformTelemetryState.instance.getSessionId(),
+                    codeTransformMavenBuildCommand = CodeTransformMavenBuildCommand.Mvnw,
+                    reason = error
+                )
             }
-            val output = runCommand(mvnw)
-            if (output.exitCode != 0) {
-                LOG.error { "mvnw command output:\n$output" }
-                val error = "The exitCode should be 0 while it was ${output.exitCode}"
+            // TODO: currently running copy dependencies even if install failed.
+            //  This Should be updated to fast fail the transform if install failed.
+            val copyOutput = runCopyCommand(mvnw)
+            if (copyOutput.exitCode != 0) {
+                LOG.error { "$mvnw command output: $copyOutput" }
+                val error = "Maven Copy: The exitCode should be 0 while it was ${copyOutput.exitCode}"
                 CodetransformTelemetry.mvnBuildFailed(
                     codeTransformSessionId = CodeTransformTelemetryState.instance.getSessionId(),
                     codeTransformMavenBuildCommand = CodeTransformMavenBuildCommand.Mvnw,
                     reason = error
                 )
-                return null
             } else {
-                LOG.warn { "mvnw executed successfully" }
+                LOG.info { "$mvnw executed successfully" }
                 shouldTryMvnCommand = false
             }
         } catch (e: ProcessNotCreatedException) {
-            val error = "./mvnw failed to execute as its likely not a unix machine"
+            val error = "$mvnw failed to execute as its likely not a unix machine"
```

Comment:
unsure that this would be the reason why -- aren't we switching between mvnw and mvnw.cmd if they're on windows?

Reply:
that's correct. we probably shouldn't even run `./mvnw` or `./mvnw.cmd` if we don't see the executable present in the root dir of the project. if we do, we can run it and catch any type of exception, I don't think we need to handle ProcessNotCreated specifically @tincheng

Reply:
Update text. Only applies to IntelliJ and not VSCode

Reply:
updated